maxon_bringup:
  ros__parameters:
    motor_names: ["motor1"]

motor1:
  ros__parameters:
    # epos's node information (must be enough to identify the node)
    device: 'EPOS4' # default: 'EPOS4'
    protocol_stack: 'CANopen' # default: 'MAXON SERIAL V2'
    interface: 'USB' # default: 'USB'
    port: 'USB0' # default: 'USB0'
    node_id: 1 # default: 0 (invalid)

    # communication settings (optional)
    # 'device' is a set of types of device, protocol_stack, interface, and port.
    # order of node initialization is that of commandline arguments.
    baudrate: 1000000 # default: 0 (keep current baudrate)
    timeout: 500 # [ms], default: 0 (keep current timeout)

    # general parameters (optional)
    use_ros_unit: true # use ros standard units (rad, rad/s, Nm) in hardware interfaces
                      # or epos standard units (quad count of encoder pulse(qc), rpm, mNm)
                      # (default: false)

    # epos's operation mode (required)
    control_mode: "profile_position"

    # encoder parameters (required)
    encoder:
      type: 1 # 1: INC 3CH, 2: INC 2CH, 4: SSI ABS BIN, 5: SSI ABS GRY
      resolution: 1024 # encoder resolution
      gear_ratio: 243 # gear ratio of motor
      inverted_polarity: false

    # profile position mode configs (optional)
    position_profile:
      velocity: 10000 # [rpm]
      acceleration: 8000 # [rpm/s]
      deceleration: 9000 # [rpm/s]